### PR TITLE
Dispatcher methods

### PR DESCRIPTION
**Pull request by imLew**
**Proposed changes**:
- Replace the `CollectingDispatcher` methods with a single method. The former structure was based on the sdk's interactin with rasa_core and is no longer necessary.

Related to https://github.com/RasaHQ/rasa/pull/4687
(Guess I should also update the changelog there)

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa-sdk#code-style) for instructions)

### CODE REVIEW

**Comment by erohmensing** [comment]
How did I never think to do this -- I love this 😃

File: rasa_sdk/executor.py [hunk 1]
```diff
@@ -23,49 +23,69 @@ def __init__(self) -> None:
...
-
-    def utter_message(self, text: Text, **kwargs: Any) -> None:
-        """"Send a text to the output channel"""
-
-        message = {"text": text}
-        message.update(kwargs)
-
-        self.messages.append(message)
+        warnings.warn(
+            "Use of `utter_elements` is deprecated. " "Use `utter_message` instead.",
```
Comment (tabergma) [inline rasa_sdk/executor.py]:
```suggestion
            "Use of `utter_elements` is deprecated. Use `utter_message` instead.",
```

File: rasa_sdk/executor.py [hunk 2]
```diff
@@ -23,49 +23,69 @@ def __init__(self) -> None:
...
+            "Use of `utter_button_message` is deprecated. "
+            "Use `utter_message` instead.",
+            DeprecationWarning,
+        )
 
-        message = {"text": text, "buttons": buttons}
-        message.update(kwargs)
-
-        self.messages.append(message)
+        self.utter_message(text=text, buttons=buttons)
```
Comment (tabergma) [inline rasa_sdk/executor.py]: What about `kwargs`? Same for the other methods.

File: rasa_sdk/executor.py [hunk 3]
```diff
@@ -23,49 +23,69 @@ def __init__(self) -> None:
...
 
     def utter_attachment(self, attachment: Text, **kwargs: Any) -> None:
         """Send a message to the client with attachments."""
+        warnings.warn(
+            "Use of `utter_attachment` is deprecated. " "Use `utter_message` instead.",
```
Comment (tabergma) [inline rasa_sdk/executor.py]:
```suggestion
            "Use of `utter_attachment` is deprecated. Use `utter_message` instead.",
```

File: rasa_sdk/executor.py [hunk 4]
```diff
@@ -77,38 +97,43 @@ def utter_button_template(
...
-        self.messages.append(message)
+        self.utter_message(template=template, buttons=buttons)
 
     # noinspection PyUnusedLocal
     def utter_template(
         self, template: Text, tracker: Tracker, silent_fail: bool = False, **kwargs: Any
     ) -> None:
         """"Send a message to the client based on a template."""
+        warnings.warn(
+            "Use of `utter_template` is deprecated. " "Use `utter_message` instead.",
```
Comment (tabergma) [inline rasa_sdk/executor.py]:
```suggestion
            "Use of `utter_template` is deprecated. Use `utter_message` instead.",
```

File: rasa_sdk/executor.py [hunk 5]
```diff
@@ -77,38 +97,43 @@ def utter_button_template(
...
-        message = {"template": template}
-        message.update(kwargs)
-
-        self.messages.append(message)
+        self.utter_message(template=template)
 
     def utter_custom_json(self, json_message: Dict[Text, Any], **kwargs: Any) -> None:
         """Sends custom json to the output channel."""
+        warnings.warn(
+            "Use of `utter_custom_json` is deprecated. " "Use `utter_message` instead.",
```
Comment (tabergma) [inline rasa_sdk/executor.py]:
```suggestion
            "Use of `utter_custom_json` is deprecated. Use `utter_message` instead.",
```

File: rasa_sdk/executor.py [hunk 6]
```diff
@@ -77,38 +97,43 @@ def utter_button_template(
...
-        json_message = {"custom": json_message}
-        json_message.update(kwargs)
-
-        self.messages.append(json_message)
+        self.utter_message(json_message=json_message)
 
     def utter_image_url(self, image: Text, **kwargs: Any) -> None:
         """Sends url of image attachment to the output channel."""
+        warnings.warn(
+            "Use of `utter_image_url` is deprecated. " "Use `utter_message` instead.",
```
Comment (tabergma) [inline rasa_sdk/executor.py]:
```suggestion
            "Use of `utter_image_url` is deprecated. Use `utter_message` instead.",
```

Comment (imLew) [reply rasa_sdk/executor.py]: Thanks for these. Sorry you had to do this 😓, I will start checking strings after black runs :/

File: rasa_sdk/executor.py (hunk 2 above)
Comment (imLew) [reply rasa_sdk/executor.py]: Thanks for catching that 👍 I will pass on the kwargs
(Out of curiosity: do you know if they are ever used?)

Comment (tabergma) [reply rasa_sdk/executor.py]: I have no idea 😄

Comment (imLew) [reply rasa_sdk/executor.py]: better safe than sorry 😂 
just made the change

**Comment by imLew** [comment]
> How did I never think to do this -- I love this 😃

When you're writing docs, but you're too lazy to write docs so you end up refactoring instead :P

**Comment by erohmensing** [comment]
> When you're writing docs, but you're too lazy to write docs so you end up refactoring instead :P

I'm gonna put you in charge of more docs! 😉

**Pull request merged**